src/config/settings.js

```diff
@@ -135,4 +146,52 @@ function parseSaveError(err) {
   return friendlierMsg;
 }
 
+// updateTable is a hash of [integer, function(settings) => void].
+//
+// It is currently empty, but if there are any changes across versions,
+// they should be done in a function that modifies the settings arg.  The main use-cases
+// are for renaming property names, correct values that are no longer valid, and removing
+// obsolete entries. The final step merges in current defaults, so we won't need an entry
+// for every version change, as most changes will get picked up from the defaults.
+//
+// For example:
+/*
+let updateTable = {
+  3: function(settings) {
+      // Implement setting change from version 3 to 4
+      if (settings.kubernetes.oldName && !settings.kubernetes.newName) {
+         settings.kubernetes.newName = settings.kubernetes.oldName;
+         delete settings.kubernetes.oldName;
+      }
+      if (settings.kubernetes.bird == "road runner") {
+        settings.kubernetes.bird == "roadrunner" // no spaces wanted
+      }
+   },
+};
+*/
+let updateTable = {
+}
+
+function updateSettings(settings) {
+  if (Object.keys(settings).length == 0) {
+    return defaultSettings;
+  }
+  let loaded_version = settings.version || 0;
+  if (loaded_version < CURRENT_SETTINGS_VERSION) {
+    for (; loaded_version < CURRENT_SETTINGS_VERSION; loaded_version++) {
+      if (updateTable[loaded_version]) {
+        updateTable[loaded_version](settings);
+      }
+    }
+  } else if (settings.version && settings.version > CURRENT_SETTINGS_VERSION) {
+    // We've loaded a setting file from the future, so some settings will be ignored.
+    // Try not to step on them.
+    // Note that this file will have an older version field but some fields from the future.
+    console.log(`Running settings version ${CURRENT_SETTINGS_VERSION} but loaded a settings file for version ${settings.version}: some settings will be ignored`);
+  }
+  return deepmerge(deepmerge(defaultSettings, settings),
+            { version: CURRENT_SETTINGS_VERSION});
```

Comment:
```suggestion
  settings = deepmerge(defaultSettings, settings);
  settings.version = CURRENT_SETTINGS_VERSION;
  return settings;
```

Reply:
Actually, I did this in commit 8bb273f. Note that because of the comparison on the result from `updateSettings`, I had to pass in a clone.